File: Python algo/Arrays/3-rotate.py
class Solution:
    def rotate(self, nums, k):
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        nums2 = [None] * length
        for i in range(0, max(0, length-k), 1):
            nums2[(i+k)%(length)] = nums[i]

        for i in range(0, max(length-k, length), 1):
            nums2[i] = nums[(i + length-k)%length]
        
        nums = nums2

File: js algo/easy/arrays/3-rotate.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
 var rotate = function(nums, k) {
    var len = nums.length;

    for (var i=nums.length - 1; i >= 0; i--) {
        nums[(i + k)] = nums[i]
    }

    for(var i = k-1; i >= 0; i--) {
        nums[i] = nums[(nums.length - k + i)]
    }
    
    nums.length = len
};

var nums = [1,2,3,4,5,6]
rotate(nums, 4)

console.log(nums)